scripts/us_census/pep/monthly_population_estimate/preprocess.py

```diff
@@ -0,0 +1,369 @@
+default_input_path = os.path.dirname(
+    os.path.abspath(__file__)) + os.sep + "input_data"
+flags.DEFINE_string("input_path", default_input_path, "Import Data File's List")
+
+
+def _extract_year(val: str) -> tuple:
+    """
+    This Methods returns true,year from the value contains year.
+    Otherwise false,''
+
+    Arguments:
+        val (str) : A string value contains data below format
+                    yyyy or yyyy [1] or .MM 1
+    Returns:
+        res (tuple) : Tuple with boolean value and year value or None
+    """
+    val = str(val).strip().split(' ', maxsplit=1)[0]
+    if val.isnumeric() and len(val) == 4:
+        return True, val
+    if val.isnumeric() and len(val) == 5:
+        return True, val[:4]
+    return False, None
+
+
+def _return_year(col: str) -> str:
+    """
+    This Methods returns year value if col contains year.
+    Otherwise pandas NA value.
+
+    Arguments:
+        col (str) : A string value contains data below format
+                    yyyy or yyyy [1] or .MM 1
+    Returns:
+        res (str) : String value with year yyyy or pandas NA value
+    """
+    res, out = _extract_year(col)
+    if res:
+        return out
+    return pd.NA
+
+
+def _return_month(col: str) -> str:
+    """
+    This Methods returns month and date value if col contains month, date.
+    Otherwise pandas NA value.
+
+    Arguments:
+        col (str) : A string value contains data below format
+                    yyyy or yyyy [1] or .MM 1
+    Returns:
+        res (str) : String value with month mm or pandas NA value
+    """
+    res = _extract_year(col)
+    if res[0]:
+        return pd.NA
+    return col
+
+
+def _year_range(col: pd.Series) -> str:
+    """
+    This method returns year range from the dataframe
+    column.
+
+    Arguments:
+        col (Series) : DataFrame Column of dtype str
+
+    Returns:
+        year_range (str) : String of Concatenated max and min year values
+    """
+    year_range = None
+    max_year = max(pd.to_datetime(col, errors='coerce').dt.year)
+    min_year = min(pd.to_datetime(col, errors='coerce').dt.year)
+    year_range = str(max_year) + '-' + str(min_year)
+    return year_range
+
+
+class CensusUSACountryPopulation:
+    """
+    CensusUSACountryPopulation class provides methods
+    to load the data into dataframes, process, cleans
+    dataframes and finally creates single cleaned csv
+    file.
+    Also provides methods to generate MCF and TMCF
+    Files using pre-defined templates.
+    """
+
+    def __init__(self, input_files: list, csv_file_path: str,
+                 mcf_file_path: str, tmcf_file_path: str) -> None:
+        self.input_files = input_files
+        self.cleaned_csv_file_path = csv_file_path
+        self.mcf_file_path = mcf_file_path
+        self.tmcf_file_path = tmcf_file_path
+        self.df = None
+        self.file_name = None
+        self.scaling_factor = 1
+
+    def _load_data(self, file: str) -> pd.DataFrame:
+        """

```

Comment:
nit (no changes necessary):  for smallish local functions, feel free to skip docstring.